tests/local.rs

```diff
@@ -353,6 +352,74 @@ async fn resolve_execute_async_value(mut c: Client, port: u16) -> Result<(), err
     Ok(())
 }
 
+async fn back_and_forward(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+
+    assert_eq!(c.current_url().await?.as_str(), sample_url);
+
+    let other_url = other_page_url(port);
+    c.goto(&other_url).await?;
+    assert_eq!(c.current_url().await?.as_str(), other_url);
+
+    c.back().await?;
+    assert_eq!(c.current_url().await?.as_str(), sample_url);
+
+    c.forward().await?;
+    assert_eq!(c.current_url().await?.as_str(), other_url);
+
+    Ok(())
+}
+
+async fn status_firefox(mut c: Client, _: u16) -> Result<(), error::CmdError> {
+    // Geckodriver only supports a single session, and since we're already in a
+    // session, it should return `false` here.
+    assert!(!c.status().await?.ready);
+    Ok(())
+}
+
+async fn status_chrome(mut c: Client, _: u16) -> Result<(), error::CmdError> {
+    // Chromedriver supports multiple sessions, so it should always return
+    // `true` here.
+    assert!(c.status().await?.ready);
+    Ok(())
+}
+
+async fn page_title(mut c: Client, port: u16) -> Result<(), error::CmdError> {
+    let sample_url = sample_page_url(port);
+    c.goto(&sample_url).await?;
+    assert_eq!(c.title().await?, "Sample Page");
+    Ok(())
+}
+
+async fn timeouts(mut c: Client, _: u16) -> Result<(), error::CmdError> {
+    let new_timeouts = TimeoutConfiguration::new(
+        Some(Duration::from_secs(60)),
+        Some(Duration::from_secs(60)),
+        Some(Duration::from_secs(30)),
+    );
+    c.set_timeouts(new_timeouts.clone()).await?;
+
+    let got_timeouts = c.get_timeouts().await?;
+    assert_eq!(got_timeouts, new_timeouts);
+
+    // Ensure partial update also works.
```

Comment:
This behavior is surprising to me. I feel like it should be called `update_timeouts` rather than `set_timeouts`?

Reply:
The spec specifically allows you to specify one or more of the params, and `TimeoutConfiguration` has `skip_serializing_if = "Option::is_none"` on all fields.

I think `update_timeouts` does fit better so I've renamed it to that.